src/common/com/intellij/plugins/haxe/ide/annotator/HaxeSemanticAnnotator.java

```diff
@@ -413,16 +442,83 @@ public void run() {
       holder.createErrorAnnotation(currentModifiers.getModifierPsi(HaxeModifierType.OVERRIDE), "Overriding nothing").registerFix(
         new RemoveModifierIntent("Remove override", currentModifiers, HaxeModifierType.OVERRIDE)
       );
-    } else if (!currentModifiers.hasModifier(HaxeModifierType.OVERRIDE) && requiredOverride) {
-      holder.createErrorAnnotation(currentMethod.getNameOrBasePsi(), "Must override").registerFix(
-        new HaxeFixer("Add override") {
-          @Override
-          public void run() {
-            currentModifiers.addModifier(HaxeModifierType.OVERRIDE);
+    }
+    else if (requiredOverride) {
+      if (!currentModifiers.hasModifier(HaxeModifierType.OVERRIDE)) {
+        holder.createErrorAnnotation(currentMethod.getNameOrBasePsi(), "Must override").registerFix(
+          new HaxeFixer("Add override") {
+            @Override
+            public void run() {
+              currentModifiers.addModifier(HaxeModifierType.OVERRIDE);
+            }
           }
-        }
+        );
+      }
+      else {
+        // It is rightly overriden. Now check the signature.
+        checkMethodsSignatureCompatibility(currentMethod, parentMethod, holder);
+      }
+    }
+  }
+
+  static public void checkMethodsSignatureCompatibility(
+    @NotNull final HaxeMethodModel currentMethod,
+    @NotNull final HaxeMethodModel parentMethod,
+    final AnnotationHolder holder
+  ) {
+    final HaxeDocumentModel document = currentMethod.getDocument();
```

Comment:
The link to the manual I put in the first comment talks about how optional arguments work -- and that they can appear before other non-optional arguments.  A little empirical testing (http://try.haxe.org/#651C8) shows excluding optional arguments to be allowed only for function calls.  For declarations and definitions (overrides and implementations), two functions in the class hierarchy can't use the same name with different signatures, and override signatures have to be identical, including the optional fields.

That all means this implementation is correct for checking declarations.  As it is, the function can't be used to check call parameters, but that's OK since it's not in the models, but in the annotator itself.